ambry-named-mysql/src/test/java/com/github/ambry/named/MySqlNamedBlobDbTest.java

```diff
@@ -164,6 +167,50 @@ public void testUpdateBlobStateToReady() throws Exception {
     assertEquals("Blob Id is not matched with the record", id, namedBlobRecord.getBlobId());
   }
 
+  @Test
+  public void testListNamedBlobs() throws Exception {
+    Calendar calendar = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+    calendar.add(Calendar.DATE, 10);
+
+    dataSourceFactory.setLocalDatacenter(foundDatacenter);
+    dataSourceFactory.triggerDataResultSet(datacenters);
+
+    final String blobNamePrefix = "blobNameForList";
+
+    // Create expired items
+    final long expiredTimeMs = calendar.getTimeInMillis() - 1;
+    final Set<String> expiredNames = new HashSet<>(Arrays.asList("name1", "name2", "name3", "name4", "name5"));
+    expiredNames.forEach(i ->
+        {
+        final NamedBlobRecord record = new NamedBlobRecord(account.getName(), container.getName(), blobNamePrefix + i, getBlobId(account, container), expiredTimeMs);
+          try {
+            namedBlobDb.put(record, NamedBlobState.READY, true).get();
+          } catch (Exception e) {
+            throw new RuntimeException(e);
+          }
+        }
+    );
+
+    // Create valid items
+    final long validimeMs = calendar.getTimeInMillis() + 8000;
+    final Set<String> validNames = new HashSet<>(Arrays.asList("name6", "name7", "name8", "name9", "name10"));
+    validNames.forEach(i ->
+        {
+          final NamedBlobRecord record = new NamedBlobRecord(account.getName(), container.getName(), blobNamePrefix + i, getBlobId(account, container), validimeMs);
+          try {
+            namedBlobDb.put(record, NamedBlobState.READY, true).get();
+          } catch (Exception e) {
+            throw new RuntimeException(e);
+          }
+        }
+    );
+
+    // List named blob should only put out valid ones without empty entries.
+    Page<NamedBlobRecord> page =  namedBlobDb.list(account.getName(), container.getName(), blobNamePrefix, null).get();
+    assertEquals("List named blob entries size should match", validNames.size(), page.getEntries().size());
```

Comment:
we can't get compare the size, we also have to compare the content of these two lists.

Reply:
Right, I find out the Unit test db query is just a mock, and it will return all rows, each row will have null for all fields (including NextPageToken). So it's not meaningful test there, I just moved to do the real test in integration test